core/plugin.go

```diff
@@ -125,12 +129,27 @@ type RequestedPlugin struct {
 	path      string
 	checkSum  [sha256.Size]byte
 	signature []byte
+	uri       *url.URL
+}
+
+// Checks if string is URL
+func isURL(url string) bool {
```

Comment:
Consider moving this function body to single source file (for example `/grpc/common/common.go`) instead of redeclaring it multiple times